File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
       
        int h1= 0;
        int h2= height.length-1;
        int size = height.length;
        int maxarea=0;
        
        while(h1<h2)
        {
            int area = Math.min(height[h1],height[h2]) * (h2-h1);  
            maxarea = Math.max(maxarea,area);
            
        if(height[h1]<height[h2])
        {
            h1++;
        }
        else
        {
            h2--;
        }
        
    }
        return maxarea;
    }
}
        
        
        
        
       

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0;
        r = len(height)-1;
        maxArea = 0;
        currArea =0;
        while l<r and r<len(height):
            currArea = (min(height[l],height[r]))*(r-l);
            maxArea = max(maxArea, currArea);
            if height[l]<height[r]:
                l = l+1;
            else:
                r=r-1;
        return maxArea;
